src/panels/lovelace/cards/hui-alarm-panel-card.ts

```diff
@@ -194,11 +196,17 @@ class HuiAlarmPanelCard extends LitElement implements LovelaceCard {
   }
 
   private _handleActionClick(e: MouseEvent): void {
+    const input = this.shadowRoot!.querySelector(
+      "#alarmCode"
+    ) as PaperInputElement;
+    const code =
```

Comment:
Why is this necessary? `_handlePadClick` is mutating `this._code` so should be up to date ?

Reply:
This is for keyboard input

Reply:
it's a bit dangerous that we keep `this._code` and the input out of sync, but fine with this.

Reply:
Yeah, considered that but thought it would be pretty unlikely and would clear after one failed attempt 🤷‍♀️